### PR TITLE
Run e2e with cluster-api provided clusters

### PR DESCRIPTION
**Pull request by aleoli**
# Description

This pr adds a new e2e test provider that creates clusters with the kubevirt cluster-api implementation

### CODE REVIEW

File: pkg/liqoctl/move/handler.go [hunk 1]
```diff
@@ -165,7 +165,7 @@ func getResticRepositoryURL(ctx context.Context, cl client.Client, isLocal bool)
 		}
 	}
 
-	return fmt.Sprintf("rest:http://%s.%s.svc.cluster.local:%d/", resticRegistry, namespace, resticPort), nil
+	return fmt.Sprintf("rest:http://%s.%s.svc:%d/", resticRegistry, namespace, resticPort), nil
```
Comment (cheina97) [inline pkg/liqoctl/move/handler.go:168]: Why is it necessary?

Comment (aleoli) [reply pkg/liqoctl/move/handler.go:168]: Yes, it creates clusters with a different domain to avoid conflicts with the main cluster's DNS resolution

**Pull request merged**